code/modules/projectiles/guns/projectile/pistol.dm

```diff
@@ -108,4 +108,33 @@
 		if(silenced)
 			user.emote("me",1,"pretends to fire a gun at [target]!")
 		else
-			user.say(pick("BANG!", "BOOM!", "PEW!", "KAPOW!"))
\ No newline at end of file
+			user.say(pick("BANG!", "BOOM!", "PEW!", "KAPOW!"))
+
+/obj/item/weapon/gun/projectile/sec
+	name = "NT Mk58 .45 pistol"
+	desc = "The NT Mk58 is a cheap, ubiquitous sidearm, produced by a NanoTrasen subsidiary. Uses .45 rounds."
+	icon_state = "secguncomp"
+	mag_type = "/obj/item/ammo_storage/magazine/c45"
+	caliber = ".45"
+	origin_tech = list(TECH_COMBAT = 2, TECH_MATERIAL = 2)
+	fire_sound = 'sound/weapons/semiauto.ogg'
+	load_method = MAGAZINE
+
+/obj/item/weapon/gun/projectile/sec/update_icon()
+	..()
+	if(stored_magazine)
+		icon_state = "secguncomp"
```

Comment:
You could reduce this to one line with

`icon_state = "secguncomp[stored_magazine ? "" : "-e"]"`

Same below on the wood variant